### PR TITLE
Remove message decorator to fix default timeouts.

### LINKED ISSUE #1145: message decorator is never getting default values from wrapped functions
I just realized of this after working with the [send_audio](https://github.com/python-telegram-bot/python-telegram-bot/blob/master/telegram/bot.py#L396) function and see that the default timeout value, 20, is not getting used. The [message](https://github.com/python-telegram-bot/python-telegram-bot/blob/master/telegram/bot.py#L68) decorator doesn't know anything about the parameters of the wrapped function and its default values. So the timeout argument and other default values are None unless is explicitly passed.

This is a problem, for example, with the [send_audio](https://github.com/python-telegram-bot/python-telegram-bot/blob/master/telegram/bot.py#L396) because big files take more than 5 seconds, and without the `timeout` argument passed to the `self._request.post` function it raise a NetworkError exception.

I submitted #1146 with the fixes

### PR DESCRIPTION
**Pull request by jsmnbom**
Closes #1145.

### CODE REVIEW

**Comment by Eldinnie** [comment]
Seems like you missed `Bot.edit_reply_markup`

**Comment by jsmnbom** [comment]
@Eldinnie I'm not sure what you mean... `Bot.edit_reply_markup` doesn't exist? and `Bot.edit_message_reply_markup` seems proper to me?

**Comment by Eldinnie** [comment]
@jsmnbom It gives an error in the logs on ci, that's as far as I looked

**Comment by jsmnbom** [comment]
Also fixes #1180 since that seemed to fit well here :)

**Comment by jsmnbom** [comment]
I need to stop using github to resolve merge conflicts... keep missing stuff. Will fix later :)

**Comment by Eldinnie** [comment]
Any fix on this incomming @jsmnbom ?

**Comment by jsmnbom** [comment]
Will fix tomorrow :)

File: telegram/bot.py [hunk 1]
```diff
@@ -132,6 +106,27 @@ def __init__(self, token, base_url=None, base_file_url=None, request=None, priva
                                                                   password=private_key_password,
                                                                   backend=default_backend())
 
+    def _message(self, url, data, **kwargs):
+        if kwargs.get('reply_to_message_id'):
+            data['reply_to_message_id'] = kwargs.get('reply_to_message_id')
```
Comment (Eldinnie) [inline telegram/bot.py]: give assignment operators now! :D
But as long as we can't use those. double `get`'s are not needed. if `kwargs.get` returns a value we can use `kwargs['item']` within the iff clause.
I have no idea if it has any change in speed or not :D

File: telegram/bot.py [hunk 2]
```diff
@@ -132,6 +106,27 @@ def __init__(self, token, base_url=None, base_file_url=None, request=None, priva
...
+
+        if kwargs.get('disable_notification'):
+            data['disable_notification'] = kwargs.get('disable_notification')
```
Comment (Eldinnie) [inline telegram/bot.py]: same as before and same goes for all below :D

File: telegram/bot.py [hunk 3]
```diff
@@ -132,6 +106,27 @@ def __init__(self, token, base_url=None, base_file_url=None, request=None, priva
...
+        if kwargs.get('reply_markup'):
+            reply_markup = kwargs.get('reply_markup')
+            if isinstance(reply_markup, ReplyMarkup):
+                data['reply_markup'] = reply_markup.to_json()
+            else:
+                data['reply_markup'] = reply_markup
+
+        result = self._request.post(url, data, timeout=kwargs.get('timeout'))
+
+        if result is True:
```
Comment (Eldinnie) [inline telegram/bot.py]: Just to be sure. You checked it's always either `True` or a `Message`?

File: tests/test_bot.py [hunk 4]
```diff
@@ -641,3 +641,16 @@ def post(*args, **kwargs):
 
         with pytest.raises(OkException):
             bot.send_photo(chat_id, open('tests/data/telegram.jpg', 'rb'), timeout=timeout)
+
+    def test_timeout_propagation_implicit(self, monkeypatch, bot, chat_id):
+        class OkException(Exception):
```
Comment (Eldinnie) [inline tests/test_bot.py]: Interesting way to test this (with the exception). I know it was already in, but why don;t we just assert in the monkeypatched method? That should work.

File: tests/test_callbackquery.py [hunk 5]
```diff
@@ -88,18 +88,20 @@ def test(*args, **kwargs):
 
     def test_edit_message_text(self, monkeypatch, callback_query):
         def test(*args, **kwargs):
+            print(args, kwargs)
```
Comment (Eldinnie) [inline tests/test_callbackquery.py]: debug line?

File: tests/test_callbackquery.py [hunk 6]
```diff
@@ -88,18 +88,20 @@ def test(*args, **kwargs):
...
             try:
+                text = args[1] == 'test'
```
Comment (Eldinnie) [inline tests/test_callbackquery.py]: if the line is in both `try` and `except` it should be outside of them.

File: telegram/bot.py (hunk 1 above)
Comment (jh0ker) [reply telegram/bot.py]: Wouldn't the most pythonic way be 

```suggestion
        if 'reply_to_message_id' in kwargs:
            data['reply_to_message_id'] = kwargs['reply_to_message_id']
```

File: telegram/bot.py (hunk 3 above)
Comment (jsmnbom) [reply telegram/bot.py]: I mean... that exact code worked before... so I would assume?

Comment (jsmnbom) [reply telegram/bot.py]: Alright, just checked, and all the methods that we use the _message helper in, do indeed return either True or a Message.

**Comment by jsmnbom** [comment]
Gave another lookover, merging with the intention of releasing a V12 beta to squash bugs.

**Pull request merged**